spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/sql/CredentialsPropertiesSetter.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2021-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spring.autoconfigure.sql;
+
+import com.google.cloud.spring.autoconfigure.core.GcpProperties;
+import com.google.cloud.spring.core.Credentials;
+import com.google.cloud.sql.CredentialFactory;
+import java.io.File;
+import java.io.IOException;
+import org.apache.commons.logging.Log;
+import org.springframework.core.io.Resource;
+
+/**
+ * Sets credentials to be used by GCP socket factory.
+ */
+final class CredentialsPropertiesSetter {
+
+  CredentialsPropertiesSetter() {
+  }
+
+  /**
+   * Set credentials to be used by the Google Cloud SQL socket factory.
+   *
+   * <p>The only way to pass a {@link CredentialFactory} to the socket factory is by passing a
+   * class name through a system property. The socket factory creates an instance of {@link
+   * CredentialFactory} using reflection without any arguments. Because of that, the credential
+   * location needs to be stored somewhere where the class can read it without any context. It could
+   * be possible to pass in a Spring context to {@link SqlCredentialFactory}, but this is a tricky
+   * solution that needs some thinking about.
+   *
+   * <p>If user didn't specify credentials, the socket factory already does the right thing by
+   * using the application default credentials by default. So we don't need to do anything.
+   */
+  static void setCredentials(GcpCloudSqlProperties sqlProperties, GcpProperties gcpProperties,
+      Log logger) {
+    Credentials credentials = null;
+
+    if (sqlProperties.getCredentials().hasKey()) {
+      // First tries the SQL configuration credential.
+      credentials = sqlProperties.getCredentials();
+    } else {
+      // Then, the global credential.
+      credentials = gcpProperties.getCredentials();
+    }
+
+    if (credentials.getEncodedKey() != null) {
```

Comment:
credentials could also be null here if neither sql nor gcp properties have them; add a check before this line to return without doing anything.

Reply:
`Credentials` is initialized (` private Credentials credentials = new Credentials();`) in sql and gcp properties so we may not run into this problem? Unless I'm missing something here.

https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/07d75000a704a2266ca4619e1b75534faec329a7/spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/sql/GcpCloudSqlProperties.java#L35

Reply:
No, that makes sense. Just looked suspicious at first glance.